pkg/execution/plugins/iamrole/types.go

```diff
@@ -14,7 +15,7 @@ const (
 )
 
 type Spec struct {
-	ServiceID             string                 `json:"serviceId,omitempty"`
+	ServiceID             voyager.ServiceName    `json:"serviceId,omitempty"`
```

Comment:
so is it a name or an ID? in JSON it's still `serviceId`...

Reply:
It needs to be name, not ID

Reply:
rename the field and vars for now and leave a TODO for renaming in JSON? or deprecate this field and add a `ServiceName` as a new field, and eventually migrate to it.

Reply:
I think we need to change these plugins to be example code rather than real code. This refactoring needs to be done in the internal codebase.